posts/2015/03/21/cdl-2015.md

```diff
@@ -0,0 +1,158 @@
+
+I was expecting more people to fail the problem, but over 50% of the scores are perfect.
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/1vo6uj3WJrGRgIV3lGFuwjpSw9BKSIY9Yjby8xWVJD64/pubchart?oid=116586824&amp;format=interactive"></iframe>
+
+The average number of submissions for a perfect problem was 11. We had a participant who submitted 124 times.
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/1i_n0RWZCyUWUhlI9eonmlL5Dhh_EZhO1Nm0QjF_f3Qs/pubchart?oid=436588329&amp;format=interactive"></iframe>
+
+The most failed test was [number 8](https://github.com/palcu/convertor/blob/master/teste/grader_test8.in). What's special about it is that it contains a single object, and most of the students assume that after the first object ending curly brace, they will have a comma.
+
+A reason why the last 3 tests have the most failing submissions is that they are really big and most of the students preferred to use `getchar` for I/O. Doing that, you overwhelm the operating system with lots of calls. A better approach is to use a buffer.
+
+Also, I've counted the time students spent solving the problem. The average time for any student was 6 hours and 15 minutes, and the average time for a student who successfully passed all the tests was 3 hours and 45 minutes. I made the assumption that if a student submitted a problem for the first time, I added an hour to his solving time, and if the difference between 2 submissions is more than 4 hours, I'm also adding one hour for his next submission. In total, students spent 40 days of time trying to get all the tests passed.
+
+## Hands-on interviews
+
+After we eliminated the students who didn't have 100 points at the problem and the students who didn't complete anything on the "What project are you most proud of?" in their application form, only 60 students remained. We sent a call-to-action to the ROSEdu community and the mentors, and 8 people replied that they could help us with the interviews.
+
+Each interview took 30 minutes. For the first part we asked them questions about the non-technical applications, then asked the applicant to talk about how he approached the problem and finally we asked him the dreaded technical questions.
+
+## Technical questions
+
+We would first ask the student to present his solution. Then, we would start asking him what would happen if we modified the problem statement and he would start to have special characters inside the keys, like parentheses or colons. Then we would ask him to tell us how easily it would have been to modify the source code and support those edge cases.
+
+From what I've observed, the shorter and cleaner the student's solution was, the easier for him was to tell us a solution.
```

Comment:
Susțin. Aș înlocui finalul cu "the easier for him was to give us a version adapted to the new requirements"